core/src/main/java/org/elasticsearch/index/query/IdsQueryBuilder.java

```diff
@@ -144,14 +149,14 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
 
     @Override
     protected IdsQueryBuilder doReadFrom(StreamInput in) throws IOException {
-        IdsQueryBuilder idsQueryBuilder = new IdsQueryBuilder(in.readStringArray());
+        IdsQueryBuilder idsQueryBuilder = new IdsQueryBuilder(in.readOptionalStringArray());
```

Comment:
this has been solved in master, no changes should be required to this class as part of this PR